test/app-tap/table.test.lua

```diff
@@ -268,6 +268,16 @@ do -- check table.equals
     local tbl_f = setmetatable({a = 15}, {__eq = function() return true end})
     test:is(table.equals(tbl_d, tbl_f), false,
             "table.equals when metatables don't match")
+    test:is(table.equals({a = false}, {a = false}), true,
+            "table.equals when booleans are used")
+    test:is(table.equals({a = false}, {}), false,
+            "table.equals when booleans are used")
+    test:is(table.equals({}, {a = false}), false,
+            "table.equals when booleans are used")
+    test:is(table.equals({a = box.NULL}, {a = false}), false,
+            "table.equals when booleans are used")
+    test:is(table.equals({a = false}, {a = box.NULL}), false,
+            "table.equals when booleans are used")
```

Comment:
> I would also add a test with {} and {a = box.NULL}.

There is such test. See line 232.

> It would be also good to extract the test case from another one (and place the issue link into the comment for it).

No problem. Fixed.